.gitignore

```diff
@@ -26,4 +26,7 @@ interfaces/
 
 mocks/
 
+# Ignore generated mocks
+*.mock.go
+
```

Comment:
I think we should check in the mocks

Reply:
I'd like to understand the reasoning behind this more. If code generation is idempotent for a given set of sources/interfaces what would the value of checking this code in be?

I can understand where we might need this to make the repo `go get`able (in the case of the go SDK), but currently this repo serves binary only distribution and not really intended to be used as a lib at this point in time.

If there are other genuine pros as well I'd be keen to chat about them, but I could see versioing this code introducing some pain points as well.

Reply:
Every golang project I have been involved with has checked in anything generated.

https://www.reddit.com/r/golang/comments/8mbi47/do_you_commit_all_the_generated_code_in_your/
https://stackoverflow.com/questions/56415527/should-a-developer-commit-go-generated-code

Basically, I think it's a good idea to do what most go developers are used to...

Reply:
I'll raise a new ticket to look at this so we can take care of the CI/CD at the same time as well as versioning the generated sources.